### PR TITLE
feat(UI): adding optional title property to featured extension

### LINKED ISSUE #6729: adding title properties to FeaturedExtension
As mention in https://github.com/containers/podman-desktop/pull/6708#issuecomment-2045124331 we need to have a text, or title inside the feature extension component.

To archive this behaivour, and make it a minimum reusable and flexible, we could introduce a `title` property to the `<FeaturedExtension/>` component, that would be used for display

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Required for https://github.com/containers/podman-desktop/issues/6215

### Screenshot / video of UI

(E.g. defining the title property to `EXTENSION`).

image

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop/issues/6729

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Comment by benoitf** [comment]
I have the feeling that the `EXTENSION` text is not aligned with the icon below

as in the mockup design
image

**Review by benoitf — APPROVED** [review]
just need confirmation of @ekidneyrh for alignment/size of the `EXTENSION` text
code and tests are 👍

**Comment by axel7083** [comment]
> just need confirmation of @ekidneyrh for alignment/size of the `EXTENSION` text code and tests are 👍

I am not sure to understand what you meant about the alignement. It is aligned on the left

image

**Comment by benoitf** [comment]
yes looks like I'm wrong for the alignment 👍 

for the text it looks bigger than the mockup

**Comment by axel7083** [comment]
> yes looks like I'm wrong for the alignment 👍
> 
> for the text it looks bigger than the mockup

I used `text-xs` if I need to make it smaller I need to set custom value.

**Comment by ekidneyrh** [comment]
Would this just be for the banner that the extension is labelled?

File: packages/renderer/src/lib/featured/FeaturedExtension.svelte [hunk 1]
```diff
@@ -7,6 +7,7 @@ import FeaturedExtensionDownload from './FeaturedExtensionDownload.svelte';
 
 export let featuredExtension: FeaturedExtension;
 export let variant: 'primary' | 'secondary' = 'primary';
+export let title: string | undefined = undefined;
```
Comment (benoitf) [inline packages/renderer/src/lib/featured/FeaturedExtension.svelte]: AFAIK the title should be provided by this component, not by the client

so maybe displayTitle: boolean = false

Comment (axel7083) [reply packages/renderer/src/lib/featured/FeaturedExtension.svelte]: Fixed, I changed to an hardcoded text hidden by default

**Comment by axel7083** [comment]
> Would this just be for the banner that the extension is labelled?

This would be for the extension banner yes

**Review by benoitf — APPROVED** [review]
thx

**Pull request merged**